File: crates/mithril-oxide-sys/cpp/Dialect/SparseTensor/IR/SparseTensor.cpp
#include "SparseTensor.hpp"

#include <mlir/Dialect/SparseTensor/IR/SparseTensor.h>


namespace mithril_oxide_sys::sparse_tensor {

const void *StorageSpecifierType_get(const void *encoding)
{
    return StorageSpecifierType::get(
        mlir::sparse_tensor::SparseTensorEncodingAttr::getFromOpaquePointer(encoding)
    ).getAsOpaquePointer();
}

} // namespace mithril_oxide_sys::sparse_tensor

File: crates/mithril-oxide-sys/src/Dialect/SparseTensor/IR/SparseTensor.rs
pub use self::ffi::DeviceAsyncTokenType_get;

#[cxx::bridge]
mod ffi {
    #[namespace = "mlir"]
    unsafe extern "C++" {
        include!("mithril-oxide-sys/cpp/Dialect/SparseTensor/IR/SparseTensor.hpp");

        type MLIRContext = crate::IR::MLIRContext::MLIRContext;
    }

    #[namespace = "mithril_oxide_sys::sparse_tensor"]
    unsafe extern "C++" {
        include!("mithril-oxide-sys/cpp/Dialect/SparseTensor/IR/SparseTensor.hpp");

        type c_void = crate::IR::Value::ffi::c_void;

        pub unsafe fn StorageSpecifierType_get(encoding: *const c_void) -> *const c_void;
    }
}
